client/src/classes/AudioVisualizer.ts

```diff
@@ -25,24 +31,24 @@ export class AudioVisualizer {
     // Feed mic. audio into the analyser.
     source.connect(analyser);
 
-    // Previous average to compare to current.
-    let prevAverage = 0;
     this.onAudioActivity(0);
-    const draw = () => {
-      const array = new Uint8Array(analyser.fftSize);
-      analyser.getByteFrequencyData(array);
-
-      const sum = array.reduce((current, next) => current + next * 4);
-      const average = sum / array.length;
-      if (isSignificantlyDifferent(prevAverage, average, 7)) {
-        this.onAudioActivity(average);
-        // Update previous average once current average meets threshold
-        prevAverage = average;
-      }
-      window.requestAnimationFrame(draw);
-    };
-    draw();
+    this.animationFrameId = window.requestAnimationFrame(this.draw(analyser, 0));
   }
+
+  draw = (analyser: AnalyserNode, prevAverage: number) => (_step: number) => {
```

Comment:
Took the draw out of the `setStream` function and placed it at the root of the class to avoid a stale closure.